src/app/routes/registration/request-claim/request-claim.component.ts

```diff
@@ -178,83 +215,59 @@ export class RequestClaimComponent implements OnInit, SubjectElements {
-          const parseVersion = (version: string | number) => {
-            if (typeof (version) === 'string') {
-              return parseInt(version.split('.')[0], 10);
-            }
-            return version;
-          };
-          const claim = {
-            fields: JSON.parse(JSON.stringify(fields)),
-            claimType: this.selectedNamespace,
-            claimTypeVersion: parseVersion(this.selectedRole.version) || DEFAULT_CLAIM_TYPE_VERSION
-          };
-
-          await this.iamService.iam.createClaimRequest({
-            issuer: did,
-            claim,
-            subject: this.roleTypeForm.value.assetDid ? this.roleTypeForm.value.assetDid : undefined
-          });
-
-        } catch (e) {
-          console.error('Enrolment Failed', e);
-          this.toastr.error(e, TOASTR_HEADER);
-          this.submitting = false;
-        } finally {
-          this.loadingService.hide();
-        }
-
-        this.displayAlert('Request to enrol as ' + this.roleTypeForm.value.roleType.name.toUpperCase() + ' is submitted for review and approval.',
-          'success');
-      } else {
-        this.toastr.error('Cannot identify issuer for this role.', TOASTR_HEADER);
+    if (did && did.length) {
+      this.submitting = true;
+      this.loadingService.show('Please confirm this transaction in your connected wallet.');
+
+      try {
+        // Submit
+        await this.iamService.iam.createClaimRequest({
+          issuer: did,
+          claim: this.createClaim(),
+          subject: this.roleTypeForm.value.assetDid ? this.roleTypeForm.value.assetDid : undefined,
+          registrationTypes: this.getRegistrationTypes()
+        } as any);
+
+      } catch (e) {
+        console.error('Enrolment Failed', e);
+        this.toastr.error(e, TOASTR_HEADER);
+        this.submitting = false;
+      } finally {
+        this.loadingService.hide();
       }
+
+      this.displayAlert('Request to enrol as ' + this.roleTypeForm.value.roleType.name.toUpperCase() + ' is submitted for review and approval.',
+        'success');
     } else {
-      this.toastr.error('Enrolment Form is invalid.', TOASTR_HEADER);
+      this.toastr.error('Cannot identify issuer for this role.', TOASTR_HEADER);
     }
-
     this.loadingService.hide();
   }
 
   goToEnrolment() {
     if (this.roleTypeForm.value.enrolFor === EnrolForType.ASSET) {
       // Navigate to My Enrolments Page
-      this.route.navigate(['dashboard'], { queryParams: { returnUrl: '/assets/enrolment/' + this.roleTypeForm.value.assetDid } });
+      this.route.navigate(['dashboard'], {queryParams: {returnUrl: '/assets/enrolment/' + this.roleTypeForm.value.assetDid}});
     } else {
       // Navigate to My Enrolments Page
-      this.route.navigate(['dashboard'], { queryParams: { returnUrl: '/enrolment?notif=myEnrolments' } });
+      this.route.navigate(['dashboard'], {queryParams: {returnUrl: '/enrolment?notif=myEnrolments'}});
```

Comment:
Haha, this is interesting. I think the formatter I use in VS Code would have added the spaces in the object. I think you've using Webstorm, right?

Reply:
Yup, I'm Webstorm fan ^^ Well i think we do not add spaces there in switchboard. However we didn't set up tslint/eslint yet, so there is no one forced way to align code. I tested VS Code but it was so different that i couldn't force myself to work with it.

Reply:
Yea, no worries, not saying you have to use VS code. Yes, we should setup eslint/prettier